### PR TITLE
Also check `USELESS_ELVIS` in `UnreachableCode`

### PR DESCRIPTION
**Pull request by atulgpt**
Fixes https://github.com/detekt/detekt/issues/6623

### CODE REVIEW

**Comment by 3flex** [comment]
I understand why this is added to the existing unreachable code rule but I believe this should be implemented as a separate rule since it's a wrapper of an existing compiler warning. I don't think wrapped compiler warnings should be consolidated in the same rule.

And very technically speaking the Elvis operator itself is not unreachable - it will always be hit.

**Comment by BraisGabin** [comment]
> I understand why this is added to the existing unreachable code rule but I believe this should be implemented as a separate rule since it's a wrapper of an existing compiler warning. I don't think wrapped compiler warnings should be consolidated in the same rule.

I'm not sure about this. It's true, it's easy to think "two different rules" looking at the implementation. But, as a user, both are unreachable code so I think that it is ok to merge them. A new rule will help to make this more configurable but I think that we don't need such configuration.

File: detekt-gradle-plugin/src/functionalTest/kotlin/io/gitlab/arturbosch/detekt/DetektAndroidSpec.kt [hunk 1]
```diff
@@ -721,7 +721,7 @@ private val SAMPLE_ACTIVITY_USING_VIEW_BINDING = """
     
         override fun onCreate(savedInstanceState: Bundle?) {
             binding = ActivitySampleBinding.inflate(LayoutInflater.from(this))
-            binding.sampleView ?: return
+            binding.sampleView
```
[the remark this answers is not in the document]
Comment (schalkms) [reply detekt-gradle-plugin/src/functionalTest/kotlin/io/gitlab/arturbosch/detekt/DetektAndroidSpec.kt:724]: I'm not sure about the return.

File: detekt-rules-errorprone/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/bugs/UnreachableCode.kt [hunk 2]
```diff
@@ -43,7 +43,9 @@ class UnreachableCode(config: Config = Config.empty) : Rule(config) {
 
     override fun visitExpression(expression: KtExpression) {
         super.visitExpression(expression)
-        if (bindingContext.diagnostics.forElement(expression).any { it.factory == Errors.UNREACHABLE_CODE }) {
+        if (bindingContext.diagnostics.forElement(expression)
+                .any { it.factory in listOf(Errors.UNREACHABLE_CODE, Errors.USELESS_ELVIS) }
```
Comment (schalkms) [inline detekt-rules-errorprone/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/bugs/UnreachableCode.kt:47]: We shouldn't create a list each and every time. For two elements it's better to check with an OR `||`.

Comment (atulgpt) [reply detekt-rules-errorprone/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/bugs/UnreachableCode.kt]: Although I personally prefer list type of usage as that scales well with more checks. But I am okay with using `||`. I have updated the code

**Review by BraisGabin — APPROVED** [review]

**Pull request merged**